### PR TITLE
feat: Improve Skia TextBoxView overlay positioning and sizing

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): related to #7514

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

See below

## What is the new behavior?

- Previously we were using LayoutChanged and SizeChanged, but that was not completely reliable. Now we do the invalidation before render, which is much more accurate
- To prevent unnecessary operations we short-circuit as soon as possible - only computing when a TextBoxView is actually focused

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Gtk/UnoDrawingArea.cs [hunk 1]
```diff
@@ -21,18 +22,24 @@ public UnoDrawingArea()
...
+			_focusManager?.FocusRectManager?.RedrawFocusVisual();
+			if (_focusManager?.FocusedElement is TextBox textBox)
 			{
-				_focusManager = VisualTree.GetFocusManagerForElement(Windows.UI.Xaml.Window.Current?.RootElement);
+				textBox.TextBoxView?.Extension?.InvalidateLayout();
 			}
-			_focusManager?.FocusRectManager?.RedrawFocusVisual();
+		}
+
+		private void InvalidateInputField()
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Gtk/UnoDrawingArea.cs:40]: is this needed?

**Review by davidjohnoliver — APPROVED** [review]
any way to unit-test these changes?

File: src/Uno.UI.Runtime.Skia.Gtk/UnoDrawingArea.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI.Runtime.Skia.Gtk/UnoDrawingArea.cs]: Nope and gone :-)

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**